### DISCUSSION TITLE
Problem configuring REST API

### DISCUSSION THREAD

**Discussion by miguelem** [Budibase/budibase#6470] [Help]
I'm trying to set up the Google Ads API as a REST data source. In the "auth" selector it only lets me select "None", but I need Oauth2. 

![image](https://user-images.githubusercontent.com/37700452/175895624-7cd2ee40-6bc0-43eb-af57-e34b2f717cfb.png)

How can I solve it?

Also, all the documentation that you have for REST APIs is in https://docs.budibase.com/docs/rest? Or do you have examples or more explanations somewhere else?

**Comment by melohagan** [comment]
HI @miguelem 

You need to add an auth option within the REST config page: https://docs.budibase.com/docs/rest-authentication

**Comment by miguelem** [comment]
OK, but I don't see an Oauth2 option. 

Is Budibase compatible with "advanced" Oauth2 systems like Google Ads API? (which needs Client ID, Client secret and a couple of tokens as identification) https://developers.google.com/google-ads/api/docs/oauth/internals

**Comment by melohagan** [comment]
@miguelem Ah I see. Support for SSO tokens from log-in within the REST connector will be available in the next release - 11th of July I believe.
https://github.com/Budibase/budibase/issues/3827

**Comment by miguelem** [comment]
I don't think SSO tokens are what I need (at least SSO is not mentioned at all in the Google Ads API documentation). I need Oauth2.

**Reply by melohagan** [reply]
I'm not super familiar with OAuth2, but I think Google's SSO uses OAuth2. So this new feature when you login, will allow you to reuse the auth token received from successful login.
